core/schema/directory.graphqls

```diff
@@ -14,76 +14,197 @@ type Directory {
+    """
+    Location of the written directory (e.g., "/src/").
+    """
+    path: String!,
+
+    "Identifier of the directory to copy."
+    directory: DirectoryID!,
+
     """
-    Exclude artifacts that match the given pattern.
-    (e.g. ["node_modules/", ".git*"]).
+    Exclude artifacts that match the given pattern (e.g., ["node_modules/", ".git*"]).
     """
-    exclude: [String!]
+    exclude: [String!],
 
     """
-    Include only artifacts that match the given pattern.
-    (e.g. ["app/", "package.*"]).
+    Include only artifacts that match the given pattern (e.g., ["app/", "package.*"]).
     """
-    include: [String!]
+    include: [String!],
   ): Directory!
 
-  "Retrieves this directory plus a new directory created at the given path."
-  withNewDirectory(path: String!, permissions: Int): Directory!
+  """
+  Retrieves this directory plus a new directory created at the given path.
+  """
+  withNewDirectory(
+    """
+    Location of the directory created (e.g., "/logs").
+    """
+    path: String!,
+
+    """
+    Permission granted to the created directory (e.g., 0777).
+
+    Default: 0755.
+    """
+    permissions: Int
+  ): Directory!
 
-  "Retrieves this directory with the directory at the given path removed."
-  withoutDirectory(path: String!): Directory!
+  """
+  Retrieves this directory with the directory at the given path removed.
+  """
+  withoutDirectory(
+    """
+    Location of the directory to remove (e.g., ".github/").
+    """
+    path: String!
+  ): Directory!
 
   "Gets the difference between this directory and an another directory."
-  diff(other: DirectoryID!): Directory!
+  diff(
+    "Identifier of the directory to compare."
+    other: DirectoryID!
+  ): Directory!
 
-  "Writes the contents of the directory to a path on the host."
-  export(path: String!): Boolean!
+  """
+  Writes the contents of the directory to a path on the host.
+  """
+  export(
+    """
+    Location of the copied directory (e.g., "logs/").
+    """
+    path: String!
+  ): Boolean!
 
-  "Builds a new Docker container from this directory."
+  """
+  Builds a new Docker container from this directory.
+  """
   dockerBuild(
     """
-    Path to the Dockerfile to use.
-    Defaults to './Dockerfile'.
+    Path to the Dockerfile to use (e.g., "frontend.Dockerfile").
+
+    Defaults: './Dockerfile'.
     """
-    dockerfile: String
+    dockerfile: String,
 
     "The platform to build."
-    platform: Platform
+    platform: Platform,
 
-    "Additional build arguments."
-    buildArgs: [BuildArg!]
+    "Build arguments to use in the build."
+    buildArgs: [BuildArg!],
 
     "Target build stage to build."
     target: String
   ): Container!
 
-  "Retrieves this directory with all file/dir timestamps set to the given time, in seconds from the Unix epoch."
-  withTimestamps(timestamp: Int!): Directory!
+  """
+  Retrieves this directory with all file/dir timestamps set to the given time, in seconds from the Unix epoch.
```

Comment:
The part "in seconds from the Unix epoch" is redundant with the `timestamp` parameter doc.

```suggestion
  Retrieves this directory with all file/dir timestamps set to the given time.
```